index.rst

```diff
@@ -2,28 +2,137 @@
+This interoperation allows us, for instance, to execute user-defined sequence of observations between scheduler-driven operations.
+
+The Scheduling Algorithm
+------------------------
+
+The Scheduler CSC itself only implements the business logic required to compute an observing queue and to execute them on the ScriptQueue.
+The actual algorithm responsible for digesting the telemetry stream and producing targets interfaces with the CSC by means of a well defined API, called the ``Driver``.
+
+For the time being the project has settle on the "Feature Based Scheduler" as the baseline application for implementing the scheduling algorithm.
+
+Integration between the Featurer Based Scheduler and the Scheduler CSC is already completed and been heavily used to drive observations with the Auxiliary Telescope.
+
+Handling ToOs
+=============
+
+The Rubin Observatory Control System design allow ToOs to be handled in a number of different ways.
+It is likely that, during operations. more than one mode of operation will be used, depending on the ToO conditions and project guidelines.
+
+For now we will focus on the two main different ways to respond to ToOs; autonomously through the Scheduler CSC and manually through the ScriptQueue.
+
+Autonomous Response With the Scheduler
+--------------------------------------
+
+In principle, in order to autonomously respond to ToOs the Scheduler CSC needs first to be configured with a Feature Based Scheduler setup that is capable or responding to inputs from a ToO alert system.
+
+The Feature Based Scheduler already has a `ToO_survey`_, that is currently successfully used for simulation purposes and that could also be used for operations.
+
+.. _ToO_survey: https://rubin-sim.lsst.io/api/rubin_sim.scheduler.surveys.ToO_survey.html#rubin_sim.scheduler.surveys.ToO_survey
+
+With that in place, the only additional element required by the Scheduler is a source of ToOs events.
+
+ToO Events as Telemetry
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Currently the Scheduler CSC colects telemetry from the EFD and hands them over to the ``Driver``, which is in charge of formatting it in a way the scheduling algorithm understands.
+By using a general purpose interface for colecting and passing telemetry from the EFD to the scheduling algorithm, it allows us to easily add new data sources as long as the data is in the EFD.
+
+It is natural then to expect that, any new information needed by the Scheduler, should also be in the EFD.
+
+By writting ToO events to the EFD the Scheduler CSC will naturally be able to access them.
+Next, all that is required is for the ``Driver`` be able to parse the information in a way the scheduling algorithm understands.
+
+This design also has the advantage that it allows us to control which ToO events the Scheduler should consider by filtering which events are written to the EFD, or adding metadata to indicate whether an event was certified or not.
+
+The diagram below summarizes the design of the system.
```

Comment:
Does the Scheduler read the EFD constantly or on prompt or some other mechanism? Trying to understand how it would actually know when a ToO lands in the EFD.